### PR TITLE
[DOC] Ensure PPL docs have consistent look

### PR DESCRIPTION
**Pull request by LantaoJin**
### Description
Main updates:
Renamed functions/ppl-lambda.md -> functions/ppl-collection.md
Move lambda functions to functions/ppl-json.md
Refactor ppl-join-command.md, ppl-lookup-command.md, ppl-subquery-command.md

### Related Issues
Resolves https://github.com/opensearch-project/opensearch-spark/issues/925

### Check List
- [x] Updated documentation (docs/ppl-lang/README.md)
- [ ] Implemented unit tests
- [ ] Implemented tests for combination with other commands
- [ ] New added source code should include a copyright header
- [x] Commits are signed per the DCO using `--signoff`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/sql/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: docs/ppl-lang/ppl-dedup-command.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
-# PPL dedup command
+## PPL dedup command
```
Comment (brijos) [inline docs/ppl-lang/ppl-dedup-command.md:1]: We are inconsistently using back ticks for commands. Let's use a back tick `dedup`

File: docs/ppl-lang/ppl-grok-command.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-## PPL Correlation Command
+## PPL Grok Command
```
Comment (brijos) [inline docs/ppl-lang/ppl-grok-command.md:1]: should be lower case and include back ticks `grok`

File: docs/ppl-lang/ppl-lookup-command.md [hunk 3]
```diff
@@ -1,20 +1,18 @@
 ## PPL Lookup Command
```
Comment (brijos) [inline docs/ppl-lang/ppl-lookup-command.md:1]: `lookup` as lowercase

File: docs/ppl-lang/ppl-top-command.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
 ## PPL top Command
```
Comment (brijos) [inline docs/ppl-lang/ppl-top-command.md:1]: missing back ticks `top`

File: docs/ppl-lang/ppl-trendline-command.md [hunk 5]
```diff
@@ -1,6 +1,6 @@
 ## PPL trendline Command
```
Comment (brijos) [inline docs/ppl-lang/ppl-trendline-command.md:1]: missing back ticks `trendline`

**Review by brijos — CHANGES_REQUESTED** [review]
Left comments around using capitalization and back ticks.

**Comment by LantaoJin** [comment]
> Left comments around using capitalization and back ticks.

Done. Thanks @brijos

**Review by YANG-DB — APPROVED** [review]

**Review by YANG-DB — APPROVED** [review]

**Pull request merged**